### PR TITLE
GDPR-10 Adding dev deployment configuration

### PR DESCRIPTION
**Pull request by brightonsbox**

### CODE REVIEW

File: helm_deploy/dps-data-compliance/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      containers:
+        - name: dps-data-compliance
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}" 
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - containerPort: {{ .Values.image.port }}
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /ping
```
Comment (petergphillips) [inline helm_deploy/dps-data-compliance/templates/deployment.yaml]: should be `/health/ping` for new services

File: helm_deploy/dps-data-compliance/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+              port: {{ .Values.image.port }}
+            periodSeconds: 30
+            initialDelaySeconds: 90
+            timeoutSeconds: 20
+            failureThreshold: 10
+          readinessProbe:
+            httpGet:
+              path: /health
```
Comment (petergphillips) [inline helm_deploy/dps-data-compliance/templates/deployment.yaml]: should be `/health/ping` for new services

File: helm_deploy/dps-data-compliance/templates/_envs.tpl [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+*/}}
+{{- define "deployment.envs" }}
+env:
+  - name: SERVER_PORT
+    value: "{{ .Values.image.port }}"
+
+  - name: JAVA_OPTS
+    value: "{{ .Values.env.JAVA_OPTS }}"
+
+  - name: APPINSIGHTS_INSTRUMENTATIONKEY
```
Comment (petergphillips) [inline helm_deploy/dps-data-compliance/templates/_envs.tpl]: will this work?  Normally set to `APPLICATION_INSIGHTS_IKEY` I think

File: helm_deploy/dps-data-compliance/templates/deployment.yaml (hunk 2 above)
Comment (petergphillips) [reply helm_deploy/dps-data-compliance/templates/deployment.yaml]: although needs to be whitelisted in the resource server

File: helm_deploy/dps-data-compliance/templates/deployment.yaml (hunk 1 above)
Comment (brightonsbox) [reply helm_deploy/dps-data-compliance/templates/deployment.yaml]: Done

File: helm_deploy/dps-data-compliance/templates/_envs.tpl (hunk 3 above)
Comment (brightonsbox) [reply helm_deploy/dps-data-compliance/templates/_envs.tpl]: It is correct as is, at least consistent with the name in AWS - this was following the offender-case-notes example

File: helm_deploy/dps-data-compliance/templates/deployment.yaml (hunk 2 above)
Comment (brightonsbox) [reply helm_deploy/dps-data-compliance/templates/deployment.yaml]: updated

**Pull request merged**